src/v/cloud_storage/tests/remote_path_provider_test.cc

```diff
@@ -85,84 +129,123 @@ TEST(TopicFromPathTest, TestTopicFromPrefixedTopicManifestPath) {
-      "e0000000/meta/kafka/tp/5_21");
+      fmt::format("{}/5_21", expected_path).c_str());
 
     auto json_path = path_provider.partition_manifest_path_json(
       test_ntp, test_rev);
     ASSERT_TRUE(json_path.has_value());
     EXPECT_STREQ(
-      json_path.value().c_str(), "e0000000/meta/kafka/tp/5_21/manifest.json");
+      json_path.value().c_str(),
+      fmt::format("{}/5_21/manifest.json", expected_path).c_str());
 
     partition_manifest pm(test_ntp, test_rev);
     EXPECT_STREQ(
       path_provider.partition_manifest_path(pm).c_str(),
-      "e0000000/meta/kafka/tp/5_21/manifest.bin");
+      fmt::format("{}/5_21/manifest.bin", expected_path).c_str());
     EXPECT_STREQ(
       path_provider.spillover_manifest_path(pm, test_spill_comps).c_str(),
-      "e0000000/meta/kafka/tp/5_21/manifest.bin.10.11.0.1.999.1000");
+      fmt::format("{}/5_21/manifest.bin.10.11.0.1.999.1000", expected_path)
+        .c_str());
     EXPECT_STREQ(
       pm.get_manifest_path(path_provider)().native().c_str(),
-      "e0000000/meta/kafka/tp/5_21/manifest.bin");
+      fmt::format("{}/5_21/manifest.bin", expected_path).c_str());
 
     spillover_manifest spill_m(test_ntp, test_rev);
     spill_m.add(test_smeta);
     EXPECT_STREQ(
       path_provider.partition_manifest_path(spill_m).c_str(),
-      "e0000000/meta/kafka/tp/5_21/manifest.bin.10.11.0.1.999.1000");
+      fmt::format("{}/5_21/manifest.bin.10.11.0.1.999.1000", expected_path)
+        .c_str());
     EXPECT_STREQ(
       spill_m.get_manifest_path(path_provider)().native().c_str(),
-      "e0000000/meta/kafka/tp/5_21/manifest.bin.10.11.0.1.999.1000");
+      fmt::format("{}/5_21/manifest.bin.10.11.0.1.999.1000", expected_path)
+        .c_str());
     partition_manifest* disguised_manifest = &spill_m;
     EXPECT_STREQ(
       disguised_manifest->get_manifest_path(path_provider)().native().c_str(),
-      "e0000000/meta/kafka/tp/5_21/manifest.bin.10.11.0.1.999.1000");
+      fmt::format("{}/5_21/manifest.bin.10.11.0.1.999.1000", expected_path)
+        .c_str());
 }
 
-TEST(RemotePathProviderTest, TestLabeledPartitionManifestPaths) {
-    remote_path_provider path_provider(test_label, test_tp_ns_override);
+TEST_P(
+  OverrideParamRemotePathProviderWithLabelTest,
+  TestLabeledPartitionManifestPaths) {
+    const auto& expected_ns = has_override ? test_tp_ns_override.ns
+                                           : test_tp_ns.ns;
+    const auto& expected_tp = has_override ? test_tp_ns_override.tp
+                                           : test_tp_ns.tp;
+    const auto expected_path = fmt::format(
+      "meta/{}/{}", expected_ns, expected_tp);
     EXPECT_STREQ(
       path_provider.partition_manifest_path(test_ntp, test_rev).c_str(),
-      "deadbeef-0000-0000-0000-000000000000/meta/kafka/tp/5_21/manifest.bin");
+      fmt::format(
+        "deadbeef-0000-0000-0000-000000000000/{}/5_21/manifest.bin",
+        expected_path)
```

Comment:
nit: as far as reading/understanding the path provider, i liked the fact that previously we could see exactly some unobstructed examples of the paths that were being generated. I know it will be more code, but would you mind writing the expected paths as plain strings? Especially for something that isn't complicated logic, IMO that kind of readability trumps reuse

Reply:
Rewrote this test with more explicit path strings.